### PR TITLE
Version update for Spark 1.5.0 and add CHANGES.txt file.

### PR DESCRIPTION
**Pull request by rxin**

### CODE REVIEW

File: ec2/spark_ec2.py [hunk 1]
```diff
@@ -51,7 +51,7 @@
     raw_input = input
     xrange = range
 
-SPARK_EC2_VERSION = "1.4.0"
+SPARK_EC2_VERSION = "1.5.0"
```
Comment (shivaram) [inline ec2/spark_ec2.py]: Also change the default branch to `branch-1.5` in this file

File: ec2/spark_ec2.py [hunk 2]
```diff
@@ -71,6 +71,7 @@
     "1.3.0",
     "1.3.1",
     "1.4.0",
+    "1.5.0"
```
Comment (shivaram) [inline ec2/spark_ec2.py]: And we need to update the spark tachyon map -- i think we use Tachyon 0.7.1 in Spark 1.5.0

File: core/src/main/scala/org/apache/spark/package.scala [hunk 3]
```diff
@@ -43,5 +43,5 @@ package org.apache
 
 package object spark {
   // For package docs only
```
Comment (vanzin) [inline core/src/main/scala/org/apache/spark/package.scala]: BTW, not your fault, and I'm not sure what exactly this comment refers to, but it's misleading. `SPARK_VERSION` is definitely not used only for package docs.

File: dev/create-release/generate-changelist.py [hunk 4]
```diff
@@ -31,8 +31,8 @@
 import traceback
 
 SPARK_HOME = os.environ["SPARK_HOME"]
-NEW_RELEASE_VERSION = "1.0.0"
-PREV_RELEASE_GIT_TAG = "v0.9.1"
+NEW_RELEASE_VERSION = "1.5.0"
+PREV_RELEASE_GIT_TAG = "v1.0.0"
```
Comment (vanzin) [inline dev/create-release/generate-changelist.py]: Shouldn't this be `v1.4.0`?

**Comment by shivaram** [comment]
LGTM. I think some of these should only be at the tag 1.5.0-rc1 or 1.5.0-rc2 etc. and not in `branch-1.5` itself. But I think you can send another PR to restore these to 1.5.1-SNAPSHOT after the RCs ?

**Comment by vanzin** [comment]
LGTM, although I haven't searched the code base to see whether any other places need the same treatment.

File: dev/create-release/generate-changelist.py (hunk 4 above)
Comment (rxin) [inline dev/create-release/generate-changelist.py]: yes -- locally I'm doing that.

**Comment by rxin** [comment]
I also added the CHANGES.txt file now.

**Pull request closed without merging**